src/elfpy/simulators.py

```diff
@@ -149,6 +153,8 @@ def set_pricing_model(self, model_name):
         else:
             raise ValueError(f'pricing_model_name must be "HyperDrive" or "Element", not {model_name}')
 
+        return self.pricing_model
```

Comment:
If it is being returned, then why is it also being assigned to `self` in the function? I think the better pattern is to make this a static method and assign the returned statement to `self.pricing_model` later.

Reply:
100% agree here, there is no reason to keep `pricing_model` in the simulator class